### DISCUSSION TITLE
Jetstream with Request-Reply

### DISCUSSION THREAD

**Discussion by earthpyy** [nats-io/nats.py#221] [Q&A] [answered]
Hi! I just want to ask if Jetstream (in `nats.py`) could be used with Request-Reply (like RPC) because I cannot see any documentation or tutorial about it. Or I just need to use the basic NATS to achieve this. Thanks!

**Comment by ripienaar** [comment, marked as answer]
Streams are for operations that are decoupled, so Request-Reply does not really make a lot of sense, more likely you would add a header with a next-stream name and stick processed data in there.  Your "request" then is a message with the header set.

**Reply by earthpyy** [reply]
I clearly see your point. BTW, thanks for the instant reply, and have a nice day!

**Reply by pckhoi** [reply]
Hmm, operations could be decoupled temporally but still coupled logically, no? The requester might not wait for a reply but still need to react to one later.

**Reply by ripienaar** [reply]
In that case put your replies into another stream.
